pkg/elfwriter/helpers.go

```diff
@@ -23,6 +23,15 @@ import (
 	"runtime/debug"
 )
 
+type zeroReader struct{}
+
+func (*zeroReader) ReadAt(p []byte, off int64) (_ int, _ error) {
+	for i := range p {
+		p[i] = 0
```

Comment:
I find it a bit odd that a reader is modifying a slice, but I guess we do this to avoid allocations, right?

Reply:
It's more to avoid using wrong bytes if the slice somehow has data.